Backend/BackendAPI/Repository/Repositories/BikeRepository.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using BackendAPI.Data;
+using BackendAPI.Models;
+using BackendAPI.Repository.Interfaces;
+
+namespace BackendAPI.Repository.Repositories
+{
+    public class BikeRepository : GenericRepository<Bike>, IBikeRepository
+    {
+        public BikeRepository(DataContext dbContext) : base(dbContext)
+        { }
+
+        public new bool Delete(int ID)
+        {
+            Bike bike = GetByID(ID);
+            if (bike == null)
+                return false;
+            dbContext.Bikes.Remove(bike);
+            return true;
+
+        }
+
+        public new IList<Bike> Get()
+        {
+            return dbContext.Bikes.ToList();
+        }
+
+        public new Bike GetByID(int ID)
+        {
+            try
+            {
+                return dbContext.Bikes.First(b => b.ID == ID);
+            }
+            catch
+            {
+                return null;
+            }
+        }
+
+        public new bool Insert(Bike component)
+        {
+            try
+            {
+                dbContext.Add(component);
+                return true;
+            }
+            catch
+            {
+                return false;
+            }
```

Comment:
Z tego co znalazłem to DbContext.Add nie rzuca wyjątków https://docs.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.dbcontext.add?view=efcore-2.1 Wyjątek chyba leci dopiero przy savechanges https://docs.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.dbcontext.savechanges?view=efcore-2.1#Microsoft_EntityFrameworkCore_DbContext_SaveChanges